spread.yaml

```diff
@@ -1017,29 +986,16 @@ suites:
             # Install the snapd built
             dpkg -i "$SPREAD_PATH"/../snapd_*.deb
 
-            #shellcheck source=tests/lib/nested.sh
-            . "$TESTSLIB/nested.sh"
-            nested_prepare_env
-            nested_create_core_vm
+            "$TESTSTOOLS"/nested-state prepare
+            "$TESTSTOOLS"/nested-state build-image core
         prepare-each: |
             tests.backup prepare
-
-            #shellcheck source=tests/lib/nested.sh
-            . "$TESTSLIB/nested.sh"
-            nested_start_core_vm
-            nested_wait_for_snap_command
+            "$TESTSTOOLS"/nested-state create-vm core
         restore-each: |
-            #shellcheck source=tests/lib/nested.sh
-            . "$TESTSLIB/nested.sh"
-            nested_destroy_vm
-
+            "$TESTSTOOLS"/nested-state remove-vm
             tests.backup restore
         restore: |
-            #shellcheck source=tests/lib/nested.sh
-            . "$TESTSLIB/nested.sh"
-            nested_destroy_vm
-            nested_cleanup_env
-
+            "$TESTSTOOLS"/nested-state restore
```

Comment:
restore doesn't seem to be calling nested_destroy_vm, is this correct?

Reply:
Right, the restore cleans up the directories, but previously the VM needs to be removed.

Reply:
Ok, I see "$TESTSTOOLS"/nested-state remove-vm (meaning nested_destroy_vm) is called by restore-each above.